test/unit/test_chain_weight_clamp.cpp

```
@@ -0,0 +1,126 @@
+    }
+
+    SUBCASE("over the maximum: saturates") {
+        // Two seeds of 2^26 each sum to 2^27, which is MEM_CHAIN_W_MAX + 1 --
+        // the smallest value the old (1<<30)-1 clamp let through to wrap, and it
+        // wrapped to 0.
+        const int over = weight_of(seeds, 2, static_cast<int64_t>(1) << 26);
+        CHECK(over == static_cast<int>(MEM_CHAIN_W_MAX));
+    }
+
+    SUBCASE("far over the maximum: still saturates, not accidentally right") {
+        // The old clamp returned (1<<30)-1 here, which stores as
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)